presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1283,8 +1293,8 @@ private void analyzeWindowFunctions(QuerySpecification node, List<Expression> ou
                     throw new SemanticException(NOT_SUPPORTED, node, "DISTINCT in window function parameters not yet supported: %s", windowFunction);
                 }
 
-                if (window.getFrame().isPresent()) {
-                    analyzeWindowFrame(window.getFrame().get());
+                if (windowSpecification.getFrame().isPresent()) {
```

Comment:
You can use imperative style instead:
`windowSpecification.getFrame().ifPresent(this::analyzeWindowFrame)`